keyboards/dztech/dz60rgb/rgbmatrix_layers.c

```diff
@@ -0,0 +1,40 @@
+#include QMK_KEYBOARD_H
+#include <rgblight_list.h>
+#include "dz60rgb.h"
+
+#define RGB_OFF 0x00, 0x00, 0x00
+
+// TODO: add proper defines so people can turn this all on and off
+// TODO: find a way to allow the user to customize colors (another enum?)
+
+
+// set weakly, can be overridden
+__attribute__ ((weak))
+void rgb_matrix_layer_helper (uint8_t red, uint8_t green, uint8_t blue, bool default_layer) {
+    for (int i = 0; i < DRIVER_LED_TOTAL; i++) {
+        if (HAS_FLAGS(g_led_config.flags[i], LED_FLAG_MODIFIER)) {
```

Comment:
I would suggest creating a const uint8_t[] of indexes directly to the LEDs that are modifier keys, then iterate over those and pass the index directly into set color. This would allow the array to be used in keymaps cleaner / easier for custom functions. Also it's just all around faster / cleaner than a keyboard agnostic pattern like this.

That is unless the rgb led config changes between all these boards.

Reply:
Do you have an example of what you mean, here?